modules/programs/brew-bundle.nix

```diff
@@ -0,0 +1,225 @@
+# Created by: https://github.com/malob
+# Inspired by: https://github.com/lccambiaghi/nixpkgs/blob/main/modules/homebrew.nix
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.brew-bundle;
+
+  brewfileSection = heading: type: entries:
+    if entries != [] then
+      "# ${heading}\n" + (concatMapStrings (name: "${type} \"${name}\"\n") entries) + "\n"
+    else "";
+
+  masBrewfileSection = entries:
+    if entries != {} then
+      "# Mac App Store apps\n" +
+      concatStringsSep "\n" (mapAttrsToList (name: id: ''mas "${name}", id: ${toString id}'') entries) +
+      "\n"
+    else "";
+
+  brewfile = pkgs.writeText "Brewfile" (
+    (brewfileSection "Taps" "tap" cfg.taps) +
+    (brewfileSection "Brews" "brew" cfg.brews) +
+    (brewfileSection "Casks" "cask" cfg.casks) +
+    (masBrewfileSection cfg.masApps) +
+    (brewfileSection "Docker contrainers" "whalebrew" cfg.whalebrews) +
+    (if cfg.extraConfig != "" then "# Extra config\n" + cfg.extraConfig else "")
+  );
+
+  brew-bundle-command =
```

Comment:
Using a list here and `optional`/`optionals` probably makes it a bit more readable.

Reply:
Good call, I update this, and also made some other changes to replace other if statements with functions like `optionalString` and `optionalAttrs`.